tiledb/cf/core.py

```diff
@@ -462,6 +406,71 @@ def meta(self) -> Optional[tiledb.Metadata]:
             return None
         return self._metadata_array.meta
 
+    def open_array(
+        self,
+        array: Optional[str] = None,
+        attr: Optional[str] = None,
+        mode: str = None,
+    ) -> tiledb.Array:
+        if mode is None:
+            mode = self._mode
+        if array is None and attr is None:
+            raise ValueError(
+                "Cannot open array. Either an array or attribute must be specified."
+            )
+        if array is None:
+            array_names = self._group_schema.arrays_with_attr(attr)
+            if array_names is None:
+                raise KeyError(f"No attribute with name '{attr}' found.")
+            if len(array_names) > 1:
+                raise ValueError(
+                    f"The array must be specified when opening an attribute that "
+                    f"exists in multiple arrays in a group. Arrays with attribute "
+                    f"'{attr}' include: {array_names}."
+                )
+            array = array_names[0]
+        tiledb_array = tiledb.open(
+            self._array_uris[array],
+            mode=self._mode,
+            key=_get_array_key(self._key, array),
+            attr=attr,
+            config=None,
+            timestamp=self._timestamp,
+            ctx=self._ctx,
+        )
+        array_key = (array, attr)
+        self._open_arrays[array_key] = tiledb_array
+        return tiledb_array
```

Comment:
Right now this will drop the reference for a currently open array if you open the same array/attr combo without closing the previous array. Maybe change `self._open_arrays` to a `defaultdict(list)` and update the `close` and `close_array` routines to close all arrays in the list?

Reply:
Okay, but I guess I assumed people would stick with the same reference. Is it more typical to open the same (array, attr) several times?

Reply:
You are right that it is an edge case, but suppose for example someone wanted to write to an array and then read from it.